primefaces/src/main/java/org/primefaces/component/splitbutton/SplitButton.java

```diff
@@ -123,6 +125,7 @@ public List<MenuElement> getElements() {
         }
         else {
             return getChildren().stream()
+                        .filter(UIComponent::isRendered)
```

Comment:
please check isInstance first, should be much faster as isRendered

Reply:
Ha, I was doubting in which order to do it. Will change it.